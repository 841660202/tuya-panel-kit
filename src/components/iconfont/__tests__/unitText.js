import { AppState } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import UnitText from '../unitText';

jest.mock('AppState', () => {
  return {
    addEventListener: (type, callback) => {},
    removeEventListener: (type, callback) => {}
  }
});

describe('UnitText Component', () => {
  it('basic render', () => {
    const component = renderer.create(<UnitText value='1' unit='℃' />).toJSON();
    expect(component).toMatchSnapshot();

    const noComponent = renderer.create(<UnitText />).toJSON();
    expect(noComponent).toMatchSnapshot();
  });
  it('should update', () => {
    const component = renderer.create(<UnitText value='1' />);
    component.update(<UnitText value='12' />);
    expect(component.root.instance.state.value).toEqual('12');
  });
  it('should set unit', () => {
    const component = renderer.create(<UnitText value='1' />);
    const { instance } = component.root;
    instance.setUnit('℃');
    expect(instance.state.unit).toEqual('℃');
  });
  it('should set value', () => {
    const component = renderer.create(<UnitText value='1' />);
    const { instance } = component.root;
    instance.setValue('34');
    expect(instance.state.value).toEqual('34');
  });
});

