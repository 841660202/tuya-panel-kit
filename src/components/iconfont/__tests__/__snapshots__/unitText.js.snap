// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitText Component basic render 1`] = `
<View
  style={
    Array [
      Object {
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "row",
      },
      null,
    ]
  }
>
  <View>
    <ARTSurfaceView
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              false,
            ],
            Object {},
          ],
          Object {
            "height": 96,
            "width": 96,
          },
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            464.258,
            650.506,
            2,
            359.81399999999996,
            573.7099999999999,
            2,
            359.81399999999996,
            684.2979999999999,
            2,
            464.258,
            761.0939999999999,
            2,
            568.702,
            761.0939999999999,
            2,
            568.702,
            32.035999999999945,
            2,
            464.258,
            32.035999999999945,
            1,
          ]
        }
        fill={
          Array [
            0,
            1,
            1,
            1,
            1,
          ]
        }
        opacity={1}
        stroke={
          Array [
            1,
            1,
            1,
            1,
          ]
        }
        strokeCap={1}
        strokeDash={
          Array [
            0,
            0,
          ]
        }
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            0.09375,
            0,
            0,
            -0.09375,
            0,
            84,
          ]
        }
      />
    </ARTSurfaceView>
  </View>
</View>
`;

exports[`UnitText Component basic render 2`] = `null`;
